Csv/CsvReader.cs

```diff
@@ -135,7 +135,11 @@ private static IEnumerable<ICsvLine> ReadImpl(TextReader reader, CsvOptions opti
                         continue;
                 }
 
-                // TODO: #11 might need to read another line if this one isn't ended yet (i.e. open quoted value)
+                // if there are an odd number of quotes, the value continues on the next line
+                while(line.ToCharArray().Count(c => c == '"') % 2 == 1)
+                {
+                    line += Environment.NewLine + reader.ReadLine();
```

Comment:
I would prefer to use the newline character that was used in the source; however it's not possible (at least I'm not aware of a way) to get that after it's been consumed by the TextReader.  Perhaps if it's important to go one way or the other it could be included in the options, something like `options.NewLine = \r\n"`.

Reply:
I would prefer adding the options for the NewLine, and an option for allowing/detecting new lines, the current code will do this for each line in each parsed file, looks like a lot of overhead.
I don't think the ToCharArray() is necessary, .Count works directly on a string as well.

Won't this give problems when a single quote is used to escape a double quote, e.g.:
a;'partly " quoted';b

I don't think we know how the line was quoted, it's part of the Regex magic.

Suggestions?

Reply:
The most difficult thing about working with CSV files is that there is no definitive standard for the format; however, the majority of CSV implementations seem to follow a handful of rules, which are well summarized here: https://tools.ietf.org/html/rfc4180#section-2

A relevant part of this pseudo-spec is as follows:

> Fields containing line breaks (CRLF), double quotes, and commas should be enclosed in double-quotes

There is no mention in this document that I can see about single quotes being a valid way to enclose a field, and in practice I have never seen single quotes used in this way.

Also, it doesn't appear that your implementation as it stands supports enclosing fields in single quotes, by examining CSVReader.cs:210-211, where you are removing the enclosing quotes:

```cs
                if (str.StartsWith("\"") && str.EndsWith("\""))
                    str = str.Substring(1, str.Length - 2).Replace("\"\"", "\"");
```

I'm happy to add the newline option, but I would respectfully push back on requiring multiline field values enclosed in single quotes.  Perhaps if you wanted to officially support that at a later time it could be a separate enhancement?

Reply:
As for the `ToCharArray()`, I received the following error when trying to not use it (even though Intellisense told me I could):
![image](https://user-images.githubusercontent.com/12286274/63060166-b3b3c200-beae-11e9-8e37-5cf666f6af81.png)

Reply:
I added those options and ended up replacing the `string.ToArray().Count()` with a `Regex.Matches().Count`, which should result in better performance as it can utilize the compiled Regex cache (https://docs.microsoft.com/en-us/dotnet/standard/base-types/compilation-and-reuse-in-regular-expressions#the-regular-expressions-cache)

Reply:
> Won't this give problems when a single quote is used to escape a double quote, e.g.: a;'partly " quoted';b

> I would respectfully push back on requiring multiline field values enclosed in single quotes

on the other hand...

> Due to lack of a single specification, there are considerable differences among implementations.  Implementors should "be conservative in what you do, be liberal in what you accept from others" (RFC 793 [8]) when processing CSV files.

In the spirit of this, I went ahead an implemented full (optional) support for both single quotes enclosing fields, as well as (optional) double quotes escaped with the backslash character (i.e. `\"`)